commit b13adc15a6ae99d8920e7dd3a836419d7b8f72cf
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2019-06-21T15:50:28-07:00 (4 months ago)
    
    SWSUP-1216 Package resize from portal, CLI, and admin-ui ignoring package traits.
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/lib/machines.js b/lib/machines.js
index 8cb4bb8e6..d18832aa1 100644
--- a/lib/machines.js
+++ b/lib/machines.js
@@ -2134,6 +2134,12 @@ function resize(req, res, next) {
                     pkgRam, imgReq.max_ram));
     }
 
+
+    if (!jsprim.deepEqual(req.pkg.traits, req.machine.package.traits)) {
+        return next(new InvalidArgumentError(
+            'Resizing to package with differing traits not supported'));
+    }
+
     var callback = updateCallback(req, res, next),
         params = {
             uuid: req.params.machine,
diff --git a/package.json b/package.json
index 4c9a230ca..f7abc4549 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "cloudapi",
     "description": "Triton CloudAPI",
-    "version": "9.6.0",
+    "version": "9.7.1",
     "author": "Joyent (joyent.com)",
     "private": true,
     "engines": {
